File: functions/approveProcessedSensorV2/lambdaInvoker.js
import { LambdaClient, InvokeCommand } from '@aws-sdk/client-lambda';

const lambdaClient = new LambdaClient({ region: 'us-east-2' });

export const invokeFingerprintAsync = async (payload) => {
  const fnName = process.env.FINGERPRINT_LAMBDA_NAME;
  if (!fnName) {
    console.log('FINGERPRINT_LAMBDA_NAME not set, skipping fingerprint invocation');
    return;
  }
  await lambdaClient.send(new InvokeCommand({
    FunctionName: fnName,
    InvocationType: 'Event',
    Payload: Buffer.from(JSON.stringify(payload)),
  }));
};

File: functions/approveProcessedSensorV2/lambda_invoker.py
import json
import os

import boto3


def invoke_fingerprint_async(payload):
    fn_name = os.environ.get("FINGERPRINT_LAMBDA_NAME")
    if not fn_name:
        print("FINGERPRINT_LAMBDA_NAME not set, skipping fingerprint invocation")
        return
    lambda_client = boto3.client("lambda", region_name="us-east-2")
    lambda_client.invoke(
        FunctionName=fn_name,
        InvocationType="Event",
        Payload=json.dumps(payload).encode(),
    )
